gvr-arcore/app/src/main/java/org/gearvrf/arcore/simplesample/SampleMain.java

```diff
@@ -0,0 +1,309 @@
+import org.gearvrf.GVRScene;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRTexture;
+import org.gearvrf.io.GVRCursorController;
+import org.gearvrf.io.GVRInputManager;
+import org.joml.Math;
+import org.joml.Matrix4f;
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+
+import java.util.EnumSet;
+import java.util.List;
+
+public class SampleMain extends GVRMain {
+    private static String TAG = "GVR_ARCORE";
+
+    private static float PASSTHROUGH_DISTANCE = 100.0f;
+    private static float AR2VR_SCALE = 100;
+
+    private ARCoreHelper mARCoreHelper;
+    private Vector3f mDisplayGeometry;
+
+    private GVRScene mVRScene;
+    private Session mARCoreSession;
+    private GVRSceneObject mARPassThroughObject;
+    private Frame mLastARFrame;
+    private ARCoreHandler mARCoreHandler;
+    private GVRSceneObject mCursor;
+    private GVRCursorController mCursorController;
+
+    /* From AR to GVR space matrices */
+    private float[] mGVRModelMatrix = new float[16];
+    private float[] mARViewMatrix = new float[16];
+    private float[] mGVRCamMatrix = new float[16];
+    private float[] mModelViewMatrix = new float[16];
+
+    public SampleMain() {}
+
+    @Override
+    public void onInit(final GVRContext gvrContext) throws Throwable {
+        super.onInit(gvrContext);
+        if (mARCoreSession == null) {
+            Log.e(TAG, "Invalid ARCore session!");
+            return;
+        }
+
+        mVRScene = gvrContext.getMainScene();
+
+        mARCoreHelper = new ARCoreHelper(gvrContext, mVRScene);
+
+        gvrContext.runOnGlThread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    onInitARCoreSession(gvrContext);
+                     /* Cursor controller */
+                    initCursorController(gvrContext);
+                } catch (CameraNotAvailableException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    private void onInitARCoreSession(GVRContext gvrContext) throws CameraNotAvailableException {
+        GVRTexture passThroughTexture = new GVRExternalTexture(gvrContext);
+
+        mARCoreSession.setCameraTextureName(passThroughTexture.getId());
+
+        // FIXME: detect VR screen aspect ratio. Using empirical 16:9 aspect ratio
+        /* Try other aspect ration whether virtual objects looks jumping ou sliding
+        during camera's rotation.
+         */
+        mARCoreSession.setDisplayGeometry(Surface.ROTATION_90 , 160, 90);
+
+        mLastARFrame = mARCoreSession.update();
+        mDisplayGeometry = configDisplayGeometry(mLastARFrame.getCamera());
+
+        mARCoreSession.setDisplayGeometry(Surface.ROTATION_90 ,
+                    (int)mDisplayGeometry.x, (int)mDisplayGeometry.y);
+
+        /* To render texture from phone's camera */
+        mARPassThroughObject = new GVRSceneObject(gvrContext, mDisplayGeometry.x, mDisplayGeometry.y,
```

Comment:
Why not use GVRCameraSceneObject instead?

Reply:
Hum, I will try this ..